tests/dump_tests/module_tests/interface_test.py

```diff
@@ -0,0 +1,192 @@
+        ddiff = DeepDiff(sort_lists(returned), sort_lists(expect), ignore_order=True)
+        assert not ddiff, ddiff
+    
+    def test_lag_type_interface(self, match_engine):
+        """
+        Scenario: Test the flow fetching objs related to LAG_TYPE iface
+        """
+        params = {Interface.ARG_NAME: "PortChannel1234", "namespace": ""}
+        m_intf = Interface(match_engine)
+        returned = m_intf.execute(params)
+        expect = create_template_dict(dbs=["CONFIG_DB", "APPL_DB", "ASIC_DB", "STATE_DB"])
+        expect["CONFIG_DB"]["keys"].extend(["PORTCHANNEL_INTERFACE|PortChannel1234", "PORTCHANNEL_INTERFACE|PortChannel1234|7.7.7.1/24"])
+        expect["APPL_DB"]["keys"].extend(["INTF_TABLE:PortChannel1234", "INTF_TABLE:PortChannel1234:7.7.7.1/24"])
+        expect["STATE_DB"]["keys"].extend(["INTERFACE_TABLE|PortChannel1234", "INTERFACE_TABLE|PortChannel1234|7.7.7.1/24"])
+        expect["ASIC_DB"]["keys"].append("ASIC_STATE:SAI_OBJECT_TYPE_ROUTER_INTERFACE:oid:0x60000000005ec")
+        ddiff = DeepDiff(sort_lists(returned), sort_lists(expect), ignore_order=True)
+        assert not ddiff, ddiff
+    
+    def test_subintf_type_interface(self, match_engine):
+        """
+        Scenario: Test the flow fetching objs related to Sub-Interface iface
+        """
+        params = {Interface.ARG_NAME: "Eth0.1", "namespace": ""}
+        m_intf = Interface(match_engine)
+        returned = m_intf.execute(params)
+        expect = create_template_dict(dbs=["CONFIG_DB", "APPL_DB", "ASIC_DB", "STATE_DB"])
+        expect["CONFIG_DB"]["keys"].extend(["VLAN_SUB_INTERFACE|Eth0.1", "VLAN_SUB_INTERFACE|Eth0.1|9.9.9.9/24"])
+        expect["APPL_DB"]["keys"].extend(["INTF_TABLE:Eth0.1", "INTF_TABLE:Eth0.1:9.9.9.9/24"])
+        expect["STATE_DB"]["keys"].extend(["INTERFACE_TABLE|Eth0.1", "INTERFACE_TABLE|Eth0.1|9.9.9.9/24"])
+        expect["ASIC_DB"]["keys"].append("ASIC_STATE:SAI_OBJECT_TYPE_ROUTER_INTERFACE:oid:0x60000000005e9")
+        ddiff = DeepDiff(sort_lists(returned), sort_lists(expect), ignore_order=True)
+        assert not ddiff, ddiff
+    
+    def test_no_interface(self, match_engine):
+        """
+        Scenario: Test the flow fetching objs related to an interface which is not present
+        """
+        params = {Interface.ARG_NAME: "Ethernet160", "namespace": ""}
+        m_intf = Interface(match_engine)
+        returned = m_intf.execute(params)
+        expect = create_template_dict(dbs=["CONFIG_DB", "APPL_DB", "ASIC_DB", "STATE_DB"])
+        expect["CONFIG_DB"]["tables_not_found"].extend(["INTERFACE"])
+        expect["APPL_DB"]["tables_not_found"].extend(["INTF_TABLE"])
+        expect["STATE_DB"]["tables_not_found"].extend(["INTERFACE_TABLE"])
+        expect["ASIC_DB"]["tables_not_found"].append("ASIC_STATE:SAI_OBJECT_TYPE_ROUTER_INTERFACE")
+        ddiff = DeepDiff(sort_lists(returned), sort_lists(expect), ignore_order=True)
+        assert not ddiff, ddiff
+    
```

Comment:
Can you add one test for invalid sub interface with an invalid vlan?

Reply:
Handled